File: OrgansDelivery.DAL/Entities/Container.cs
using OrganStorage.DAL.Interfaces;

namespace OrganStorage.DAL.Entities;

public class Container : IEntity, IMustHaveTenant, IWithName, IWithDescription
{
    public Guid Id { get; set; }
    public Guid TenantId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    
    public Organ Organ { get; set; }
    
    public Guid DeviceId { get; set; }
    public Device Device { get; set; }
}

public class ContainerRef
{
	public Guid Id { get; set; }
	public string Name { get; set; }
	public string Description { get; set; }
    public Guid OrganId { get; set; }
	public Guid DeviceId { get; set; }
}

public class ContainerDto
{
	public Guid Id { get; set; }
	public string Name { get; set; }
	public string Description { get; set; }
	public OrganRef Organ { get; set; }
	public DeviceRef Device { get; set; }
}

public class ContainerFormValues
{
    public string Name { get; set; }
    public string Description { get; set; }
	public Guid DeviceId { get; set; }
}

File: OrgansDelivery.Web/client-app/src/app/models/container.ts
import { IEntity } from "./common";
import { DeviceRef } from "./device";
import { OrganRef } from "./organ";

export interface Container extends IEntity {
    name: string;
    description: string;
    device: DeviceRef;
    organ?: OrganRef;
}

export interface ContainerRef extends IEntity {
    name: string;
    description: string;
    deviceId: string;
}

export interface ContainerFormValues {
    name: string;
    description: string;
    deviceId: string;
}
